### PR TITLE
#RI-4076-4077-4078-4079 - add bigSet, redisVersion, searchJSON and RT…

### PR DESCRIPTION
**Pull request by AmirAllayarovSofteq**
…S recommendations

### CODE REVIEW

File: redisinsight/api/src/modules/database-recommendation/scanner/strategies/redis-version.strategy.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { convertRedisInfoReplyToObject } from 'src/utils';
+
+const minRedisVersion = '6';
+
+export class RedisVersionStrategy extends AbstractRecommendationStrategy {
+  /**
+   * Check redis version recommendation
+   * @param client
+   */
+
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/database-recommendation/scanner/strategies/redis-version.strategy.ts:15]: extra space

File: redisinsight/api/src/modules/database-recommendation/scanner/strategies/redis-version.strategy.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    client,
+  ): Promise<boolean> {
+    try {
+      const info = convertRedisInfoReplyToObject(
+        await client.sendCommand(
+          new Command('info', ['server'], { replyEncoding: 'utf8' }),
+        ) as string,
+      );
+      const version = get(info, 'server.redis_version');
+      return !(semverCompare(version, minRedisVersion) >= 0);
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/database-recommendation/scanner/strategies/redis-version.strategy.ts:26]: \>= are you sure?

File: redisinsight/api/src/modules/database-recommendation/scanner/strategies/rts.strategy.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    if ((timestampMemberNames.length / members.length) * 100 > maxPercentage) {
+      return true;
+    }
+    const timestampMemberScores = members.filter(({ score }) => checkTimestamp(String(score)));
+    if ((timestampMemberScores.length / members.length) * 100 > maxPercentage) {
+      return true;
+    }
+    return false;
+  }
+}
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/database-recommendation/scanner/strategies/rts.strategy.ts:28]: Well you don't have try {} catch block here when here https://github.com/RedisInsight/RedisInsight/pull/1864/files#diff-bdb486bf8c337af570bb02737c4de4cab46efb99889cde98fb4801a9c0b9287fR44 you have.
I thought that you have try catch on another level, where isRecommendationReached invoked. If you don't - please add it on the top level (most ptobably on provider or service level) to not care about it in all strategies.

File: redisinsight/api/src/modules/database-recommendation/scanner/utils/timestamp.spec.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  { input: '-1234567891', expected: false },
+  { input: 'inf', expected: false },
+  { input: '-inf', expected: false },
+];
+
+describe('checkTimestamp', () => {
+  test.each(checkTimestampTests)('%j', ({ input, expected }) => {
+    expect(checkTimestamp(input)).toEqual(expected)
+  })
+})
```
Comment (arthosofteq) [inline redisinsight/api/src/modules/database-recommendation/scanner/utils/timestamp.spec.ts:28]: there is no empty line in the end

File: redisinsight/api/src/utils/recommendation-helper.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { REDISEARCH_MODULES } from 'src/constants';
+import { AdditionalRedisModule } from 'src/modules/database/models/additional.redis.module';
+
+export const isRedisearchModule = (modules: AdditionalRedisModule[]): boolean =>
+  modules?.some(({ name }) =>
+    REDISEARCH_MODULES.some((search) => name === search))
```
Comment (arthosofteq) [inline redisinsight/api/src/utils/recommendation-helper.ts:6]: the same

**Review by arthosofteq — APPROVED** [review]
Please pay attention on try {} catch
You can merge this after fix

File: redisinsight/api/src/modules/database-recommendation/scanner/strategies/redis-version.strategy.ts (hunk 2 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/database-recommendation/scanner/strategies/redis-version.strategy.ts:26]: version have to be less then 6. So yep, it is correct condition, no?

File: redisinsight/api/src/modules/database-recommendation/scanner/strategies/rts.strategy.ts (hunk 3 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/database-recommendation/scanner/strategies/rts.strategy.ts:28]: I will put it in scanner level
```
try {
      const isRecommendationReached = await strategy.isRecommendationReached(data);

      if (isRecommendationReached) {
        return { name };
      }
    } catch (err) {
      // ignore errors
      return null;
    }
   ```

**Pull request merged**